src/pkg/leaderElection.go

```diff
@@ -43,8 +44,26 @@ func RunLeaderElection(client *clientset.Clientset, lockName, lockIdentity, lock
 				log.Info().Msgf("Perform Migration")
 				for {
 					log.Info().Msgf("leader is %s", lockIdentity)
-					log.Info().Msgf("Getting replica count...")
-					time.Sleep(5 * time.Second)
+					replicaCount, err := getReplicas()
+					if err != nil {
+						log.Fatal().Msgf("Failed to get replica count")
+					}
+					log.Info().Msgf("Setting replica count to %v", replicaCount)
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						deploymentsClient := client.AppsV1().Deployments(lockNamespace)
```

Comment:
Still need to figure out what to do about api version here.  Looked into a bit but no obvious answer yet

Reply:
Ahh ya - we should discuss this.  I've done alot of work here - there are 2 options.